### PR TITLE
JBPM-4862 TaskService.getTasksAssignedAsBusinessAdministrator() doesn't respect Groups of a User

### PR DESCRIPTION
**Pull request by mswiderski**
depends on https://github.com/droolsjbpm/droolsjbpm-knowledge/pull/111

### CODE REVIEW

**Comment by mswiderski** [comment]
jenkins retest this

**Comment by mswiderski** [comment]
jenkins retest this

**Comment by mswiderski** [comment]
any comments or questions or is it good to be merged?

File: jbpm-human-task/jbpm-human-task-jpa/src/main/resources/META-INF/Taskorm.xml [hunk 1]
```diff
@@ -52,12 +52,11 @@
                 t.taskData.deploymentId,
                 t.taskData.skipable              )
             from
-            TaskImpl t,
-            OrganizationalEntityImpl businessAdministrator
+            TaskImpl t
+            join t.peopleAssignments.businessAdministrators businessAdministrators
```
Comment (mrietveld) [inline jbpm-human-task/jbpm-human-task-jpa/src/main/resources/META-INF/Taskorm.xml]: +1 on using joins! :)

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/RuntimeDataServiceImpl.java [hunk 2]
```diff
@@ -700,6 +700,7 @@ public UserTaskInstanceDesc getTaskById(Long taskId) {
 		Map<String, Object> params = new HashMap<String, Object>();
         params.put("userId", userId);
         params.put("status", allActiveStatus);
+        params.put("groupIds", identityProvider.getRoles());
```
Comment (mrietveld) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/RuntimeDataServiceImpl.java]: Is there logic later on that deals with this value possibly being null?

Comment (mswiderski) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/RuntimeDataServiceImpl.java]: roles should never be null as you need to be at least in one role to be able to use the system

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java [hunk 3]
```diff
@@ -80,6 +79,19 @@ public void testGetTasksAssignedAsBusinessAdministratorWithUserLangOneTask() {
...
+    public void testGetTasksAssignedAsBusinessAdministratorWithUserOfGroupLangOneTask() {
+        // JBPM-4862
+        String str = "(with (new Task()) { priority = 55, taskData = (with( new TaskData()) { } ), ";
+        str += "peopleAssignments = (with ( new PeopleAssignments() ) { businessAdministrators = [new Group('Crusaders')], }),";
+        str += "name = 'This is my task name' })";
+        Task task = TaskFactory.evalTask(new StringReader(str));
+        taskService.addTask(task, new HashMap<String, Object>());
+        List<TaskSummary> tasks = taskService.getTasksAssignedAsBusinessAdministrator("Bobba Fet", "en-UK");
+        assertEquals(1, tasks.size());
+    }
```
Comment (mrietveld) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java]: Sorry, could you add a "failure" case, just to be sure? Sorry for the late response..

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/RuntimeDataServiceImpl.java (hunk 2 above)
Comment (mrietveld) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/RuntimeDataServiceImpl.java]: Thanks.

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java (hunk 3 above)
Comment (mswiderski) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/TaskQueryServiceBaseTest.java]: sure thing, will add another test and push, thanks for review

**Pull request closed without merging**